File: src/main/java/com/taw/polybank/controller/atm/TransactionFilterLucia.java
package com.taw.polybank.controller.atm;

import java.sql.Date;

public class TransactionFilterLucia {
    private Date timestampBegin;
    private Date timestampEnd;
    private String transactionOwner;
    private String beneficiaryIban;
    private double amount;

    public TransactionFilterLucia(Date timestampBegin, Date timestampEnd, String transactionOwner, String beneficiaryIban, double amount){
        this.timestampBegin = timestampBegin;
        this.timestampEnd = timestampEnd;
        this.transactionOwner = transactionOwner;
        this.beneficiaryIban = beneficiaryIban;
        this.amount = amount;
    }


    public Date getTimestampBegin() {
        return timestampBegin;
    }

    public Date getTimestampEnd() {
        return timestampEnd;
    }

    public String getTransactionOwner() {
        return transactionOwner;
    }

    public String getBeneficiaryIban() {
        return beneficiaryIban;
    }

    public double getAmount() {
        return amount;
    }

    public void setTimestampBegin(Date timestampBegin) {
        this.timestampBegin = timestampBegin;
    }

    public void setTimestampEnd(Date timestampEnd) {
        this.timestampEnd = timestampEnd;
    }

    public void setTransactionOwner(String transactionOwner) {
        this.transactionOwner = transactionOwner;
    }

    public void setBeneficiaryIban(String beneficiaryIban) {
        this.beneficiaryIban = beneficiaryIban;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }
}

File: src/main/java/com/taw/polybank/controller/atm/TransactionFilterLucia.kt
package com.taw.polybank.controller.atm

import java.sql.Date

/**
 * @author Lucía Gutiérrez Molina
 */
class TransactionFilterLucia(
  @JvmField var timestampBegin: Date?,
  @JvmField var timestampEnd: Date?,
  @JvmField var transactionOwner: String?,
  @JvmField var beneficiaryIban: String?,
  private var amountString: String?,
) {
  var amount: Double?
    private set

  init {
    this.amount = (if (amountString == null || amountString == "") "0" else amountString)!!.toDouble()
  }

  fun setAmountString(amount: String?) {
    if (amount == null || amount == "") {
      this.amountString = "0.0"
      this.amount = 0.0
    } else {
      this.amountString = amount
      this.amount = amount.toDouble()
    }
  }
}
